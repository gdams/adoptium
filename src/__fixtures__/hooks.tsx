import {
  Binary,
  ContributorApiResponse,
  Contributor,
  LatestTemurin,
  MarketplaceRelease,
  News,
  ReleaseAsset,
  TemurinReleases,
} from '../hooks';

export const createRandomContributorApiData = (): ContributorApiResponse => ({
  login: 'login_mock',
  id: 0,
  node_id: 'node_id_mock',
  avatar_url: 'avatar_url_mock',
  gravatar_id: 'gravatar_id_mock',
  url: 'url_mock',
  html_url: 'html_url_mock',
  followers_url: 'followers_url_mock',
  following_url: 'following_url_mock',
  gists_url: 'gists_url_mock',
  starred_url: 'starred_url_mock',
  subscriptions_url: 'subscriptions_url_mock',
  organizations_url: 'organizations_url_mock',
  repos_url: 'repos_url_mock',
  events_url: 'events_url_mock',
  received_events_url: 'received_events_url_mock',
  type: 'type_mock',
  site_admin: false,
  contributions: 3,
});

export const createRandomTemurinRelease = (installer, id): ReleaseAsset => ({
  os: 'os_mock',
  architecture: 'arch_mock',
  release_link: new URL('https://release_link_mock'),
  platform_name: `platform_name_mock${id}`,
  release_name: 'release_name_mock',
  release_date: new Date(Date.UTC(2020, 0, 1)),
  binaries: [
    {
      type: 'type_mock',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      size: 0,
      extension: 'extension_mock',
      installer_link: installer ? new URL('https://installer_link_mock') : undefined,
      installer_checksum: installer ? 'installer_checksum_mock' : undefined,
      installer_size: installer ? 0 : undefined,
      installer_extension: installer ? 'installer_extension_mock' : undefined,
    },
  ],
})

export const createRandomMarketplaceRelease = (installer): MarketplaceRelease => ({
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  binary: {
    os: 'os_mock',
    architecture: 'arch_mock',
    distribution: 'distribution_mock',
    image_type: 'type_mock',
    jvm_impl: 'jvm_impl_mock',
    timestamp: new Date(Date.UTC(2020, 0, 1)),
    scm_ref: 'scm_ref_mock',
    openjdk_scm_ref: 'openjdk_scm_ref_mock',
    aqavit_results_link: new URL('https://aqavit_results_link_mock'),
    tck_affidavit_link: new URL('https://tck_affidavit_link_mock'),
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      sha265sum: 'sha265sum_mock',
      sha256sum_link: new URL('https://sha256sum_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
    },
    installer: installer
      ? [
          {
            name: 'installer_name_mock.msi',
            link: new URL('https://installer_link_mock'),
            sha265sum: 'installer_sha265sum_mock',
            sha256sum_link: new URL('https://installer_sha256sum_link_mock'),
            signature_link: new URL('https://installer_signature_link_mock'),
          },
        ]
      : undefined,
  },
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  }
})

export const mockLatestTemurin = (installer): LatestTemurin => ({
  download_count: 0,
  id: 'id_mock',
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'type_mock',
      heap_size: 'heap_size_mock',
      download_count: 0,
      jvm_impl: 'jvm_impl_mock',
      package: {
        name: 'name_mock.tar.gz',
        link: new URL('https://link_mock'),
        checksum: 'sha265sum_mock',
        checksum_link: new URL('https://sha256sum_link_mock'),
        metadata_link: new URL('https://metadata_link_mock'),
        signature_link: new URL('https://signature_link_mock'),
        size: 0,
        download_count: 0,
      },
      installer: installer
      ?
        {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_sha265sum_mock',
          checksum_link: new URL('https://installer_sha256sum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
    }
  ]
})

export const createRandomTemurinReleases = (installer, id): TemurinReleases => ({
  release_name: `release_name_mock_${id}`,
  release_link: new URL('https://release_link_mock'),
  source_url: new URL('https://source_url_mock'),
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  platforms: {
    'platform_mock': {
      assets: [
        {
          os: 'os_mock',
          architecture: 'architecture_mock',
          type: 'type_mock',
          link: new URL('https://link_mock'),
          checksum: `checksum_mock${id}`,
          size: 0,
          extension: 'extension_mock',
          installer_link: installer ? new URL('https://installer_link_mock') : undefined,
          installer_checksum: installer ? 'installer_checksum_mock' : undefined,
          installer_size: installer ? 0 : undefined,
          installer_extension: installer ? 'installer_extension_mock' : undefined,
        },
      ],
    },
  },
})

export const createRandomContributorViewData = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 0,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});

export const createRandomLatestForOSData = (): Binary => ({
  release_name: 'release_name_mock',
  link: new URL('https://link_mock'),
});

export const createRandomNewsAndEventsData = (): News => ({
  news: [
    {
      id: 'id_mock',
      title: 'title_mock',
      body: 'body_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
      link: 'link_mock',
    },
  ],
  events: [
    {
      id: 'id_mock',
      title: 'title_mock',
      infoLink: 'infoLink_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
    },
  ]
});

export const createRandomContributorViewData1Contribution = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 1,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});