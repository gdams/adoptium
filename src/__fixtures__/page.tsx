import {
    AsciidocPage,
    File,
} from '../types';

export const createAsciidocData = (): {
    asciidoc: AsciidocPage;
    file: File;
  } => ({
    asciidoc: {
        id: 'asciidoc-1',
        html: '<p>Hello World, this is a <pre><code>test</code></pre></p>',
        document: {
            title: 'Asciidoc Page title',
            main: 'Asciidoc Page content',
        },
        pageAttributes: {
            authors: 'author1, author2',
        },
        fields: {
            slug: '/asciidoc/asciidoc-page-title',
        }
    },
    file: {
        relativePath: 'test.adoc',
    }
});
