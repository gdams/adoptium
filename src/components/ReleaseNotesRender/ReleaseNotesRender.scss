.MuiTablePagination-toolbar p {
  margin: 0 !important;
}
