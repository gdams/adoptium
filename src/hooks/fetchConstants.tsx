import { useEffect, useState } from 'react';
import axios from 'axios';

const baseUrl = 'https://staging-api.adoptium.net/v3';

export function fetchOses(isVisible: boolean): String[] {

    const [oses, setOses] = useState<String[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/info/available_operating-systems`;

            axios.get(url)
                .then(function (response) {
                    setOses(response.data);
                })
                .catch(function (error) {
                    setOses([]);
                });
        })();
        }
    }, [isVisible]);

    return oses;
}

export function fetchArches(isVisible: boolean): String[] {

    const [arches, setArches] = useState<String[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/info/available_architectures`;

            axios.get(url)
                .then(function (response) {
                    const results:String[] = response.data;
                    const idx = results.findIndex(it => it === 'x32');
                    if(idx >= 0) results.splice(idx, 1, 'x86');
                    setArches(results);
                })
                .catch(function (error) {
                    setArches([]);
                });
        })();
        }
    }, [isVisible]);

    return arches;
}
