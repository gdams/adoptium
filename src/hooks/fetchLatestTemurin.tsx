import { useEffect, useState } from 'react';

const baseUrl = 'https://api.adoptium.net/v3';

export function fetchLatestForOS(
    isVisible: boolean,
    version: number,
    os: string,
    arch: string
): Binary | null {
    if (!os) {
        return null
    }
    const [binary, setBinary] = useState<Binary | null>(null);
    useEffect(() => {
        if (isVisible) {
        (async () => {
            setBinary(await fetchLatestForOSRequest(version, os, arch));
        })();
        }
    }, [isVisible]);

    return binary;
}

async function fetchLatestForOSRequest(version: number, os: string, arch: string) {
    const url = `${baseUrl}/assets/feature_releases/${version}/ga?os=${os}&architecture=${arch}&image_type=jdk&jvm_impl=hotspot&page_size=1&vendor=eclipse`;
    const response = await fetch(url);
    const json = (await response.json())[0];
    let binary_link = json.binaries[0].package.link
    if (json.binaries[0].installer) {
        binary_link = json.binaries[0].installer.link
    }
    return {
        release_name: json.release_name,
        link: binary_link
    };
}

export interface Binary {
  release_name: string;
  link: URL;
}