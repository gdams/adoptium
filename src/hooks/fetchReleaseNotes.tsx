import { VersionMetaData } from '.';
import { useEffect, useState } from 'react';
import axios from 'axios';

const baseUrl = 'https://api.adoptium.net/v3/info/release_notes';

export function fetchReleaseNotesForVersion(
    isVisible: boolean,
    version: any,
    sortReleaseNotesByCallback?: Function,
): ReleaseNoteDataBag | null {
    if (!version) {
        return null
    }

    const [releaseNotes, setReleaseNotes] = useState<ReleaseNoteDataBag | null>(null);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/${version}`;

            await axios.get(url.toString())
                .then(function (response) {
                    let result = response.data;
                    if(sortReleaseNotesByCallback) sortReleaseNotesByCallback(result);
                    let releaseNoteDataBag: ReleaseNoteDataBag = {
                        releaseNoteAPIResponse: result, 
                        isValid: (result.release_notes !== null)
                    };
                    setReleaseNotes(releaseNoteDataBag);
                })
                .catch(function (error) {
                    let releaseNoteDataBag: ReleaseNoteDataBag = {
                        isValid: false
                    };
                    setReleaseNotes(releaseNoteDataBag);
                });
        })();
        }
    }, [isVisible, version]);

    return releaseNotes;
}

export interface ReleaseNoteDataBag {
    releaseNoteAPIResponse?: ReleaseNoteAPIResponse
    isValid: boolean
}

export interface ReleaseNoteAPIResponse {
    id: string;
    release_name: string;
    release_notes: ReleaseNote[];
    vendor: string;
    version_data: VersionMetaData;
}

export interface ReleaseNote {
    id: string;
    link: URL;
    title: string;
    backportOf?: string;
    priority?: string;
    component?: string;
    subcomponent?: string;
    type?: string;
}
