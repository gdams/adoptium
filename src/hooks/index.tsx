export * from './useOnScreen';
export * from './useAdoptiumContributorsApi';
export * from './fetchMarketplace';
export * from './fetchTemurinReleases';
export * from './fetchTemurinArchive';
export * from './fetchLatestTemurin';
export * from './fetchNews';
export * from './fetchReleaseNotes';

export interface VersionMetaData {
    major: number;
    minor: number;
    security: number;
    pre?: string;
    patch?: number;
    build: number;
    optional?: string;
    adopt_build_number?: number;
    openjdk_version: string;
    semver: string;
};

/**
 * This method compares two versions and returns:
 * ->  0 if versions are equals
 * -> -1 if versionA is lower than versionB
 * ->  1 if versionA is highter than versionB
 * 
 * @param versionA 
 * @param versionB 
 * @returns 
 */
export function compareVersions(versionA: VersionMetaData, versionB: VersionMetaData) : number {
    let comparison = versionA.major - versionB.major;
    if (comparison === 0) {
        comparison = versionA.minor - versionB.minor;
        if (comparison === 0) {
            comparison = versionA.security - versionB.security;
            if (comparison === 0) {
                comparison = versionA.build - versionB.build;
                if (comparison === 0) {
                    comparison = (versionA.adopt_build_number ? versionA.adopt_build_number : 0) - (versionB.adopt_build_number? versionB.adopt_build_number : 0);
                }
            }
        }
    }
    return comparison;
}

export function getVersionAsString(version: VersionMetaData) : string {
    let result = `${version.major}.${version.minor}.${version.security}`;
    if(version.build) {
        result += `+${version.build}`;
        if(version.adopt_build_number) result += `.${version.adopt_build_number}`;
    }
    return result;
}
