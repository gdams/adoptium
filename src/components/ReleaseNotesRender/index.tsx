import React, { useRef, MutableRefObject } from 'react';
import { useTheme } from '@mui/material/styles';
import { useLocation } from '@reach/router';
import queryString from 'query-string';
import Box from '@mui/material/Box';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableFooter from '@mui/material/TableFooter';
import TablePagination from '@mui/material/TablePagination';
import TableRow from '@mui/material/TableRow';
import IconButton from '@mui/material/IconButton';
import { BiLastPage, BiFirstPage, BiChevronLeft, BiChevronRight } from 'react-icons/bi';

import { fetchReleaseNotesForVersion, useOnScreen } from '../../hooks';
import './ReleaseNotesRender.scss';

interface TablePaginationActionsProps {
  count: number;
  page: number;
  rowsPerPage: number;
  onPageChange: (
    event: React.MouseEvent<HTMLButtonElement>,
    newPage: number,
  ) => void;
}

function TablePaginationActions(props: TablePaginationActionsProps) {
  const theme = useTheme();
  const { count, page, rowsPerPage, onPageChange } = props;

  const handleFirstPageButtonClick = (
    event: React.MouseEvent<HTMLButtonElement>,
  ) => {
    onPageChange(event, 0);
  };

  const handleBackButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    onPageChange(event, page - 1);
  };

  const handleNextButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    onPageChange(event, page + 1);
  };

  const handleLastPageButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    onPageChange(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1));
  };

  return (
    <Box sx={{ flexShrink: 0, ml: 2.5 }}>
      <IconButton
        onClick={handleFirstPageButtonClick}
        disabled={page === 0}
        aria-label="first page"
      >
        {theme.direction === 'rtl' ? <BiLastPage /> : <BiFirstPage />}
      </IconButton>
      <IconButton
        onClick={handleBackButtonClick}
        disabled={page === 0}
        aria-label="previous page"
      >
        {theme.direction === 'rtl' ? <BiChevronRight /> : <BiChevronLeft />}
      </IconButton>
      <IconButton
        onClick={handleNextButtonClick}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="next page"
      >
        {theme.direction === 'rtl' ? <BiChevronLeft /> : <BiChevronRight />}
      </IconButton>
      <IconButton
        onClick={handleLastPageButtonClick}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="last page"
      >
        {theme.direction === 'rtl' ? <BiFirstPage /> : <BiLastPage />}
      </IconButton>
    </Box>
  );
}

const ReleaseNotesRender = (): null | JSX.Element => {
  const version = queryString.parse(useLocation().search, {decode: false}).version;

  const ref = useRef<HTMLDivElement | null>(null);
  const isVisible = useOnScreen(ref as MutableRefObject<Element>, true);
  const releaseNotes = fetchReleaseNotesForVersion(isVisible, version);
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(20);

  const handleChangePage = (
    event: React.MouseEvent<HTMLButtonElement> | null,
    newPage: number,
  ) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  return (
	  <div ref={ref} className='text-center'>
    <h2>{version}</h2>
      {releaseNotes ? (
        <TableContainer className='py-3'>
          <Table className='table table-striped' sx={{ minWidth: 650 }} aria-label='Release Notes Table'>
            <TableHead className='table-dark'>
              <TableRow className='table-head'>
                <TableCell align="left" className='fw-bold text-light'>Issue</TableCell>
                <TableCell align="left" className='fw-bold text-light'>Component</TableCell>
                <TableCell align="left" className='fw-bold text-light'>Priority</TableCell>
                <TableCell align="left" className='fw-bold text-light'>Title</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {(rowsPerPage > 0
                ? releaseNotes.release_notes.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                : releaseNotes.release_notes
              ).map((issue) => (
                <TableRow
                  key={issue.id}
                  sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
                >
                  <TableCell className='text-nowrap' component='th' scope='row'>
                    <a target='_blank' rel='noopener noreferrer' href={issue.link.toString()}>{issue.id}</a>
                  </TableCell>
                  <TableCell>{issue.subcomponent}</TableCell>
                  <TableCell>{issue.priority}</TableCell>
                  <TableCell sx={{maxWidth: '500px'}}>
                    {issue.title}
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
            <TableFooter>
              <TableRow>
                <TablePagination
                  rowsPerPageOptions={[20, 50, 75, { label: 'All', value: -1 }]}
                  colSpan={4}
                  count={releaseNotes.release_notes.length}
                  rowsPerPage={rowsPerPage}
                  page={page}
                  SelectProps={{
                    inputProps: {
                      'aria-label': 'rows per page',
                    },
                    className: 'm-0',
                    native: true,
                  }}
                  onPageChange={handleChangePage}
                  onRowsPerPageChange={handleChangeRowsPerPage}
                  ActionsComponent={TablePaginationActions}
                />
              </TableRow>
            </TableFooter>
          </Table>
        </TableContainer>
      ) : (
        <>
        <h2>Oops... We couldn't find those release notes</h2>
        <span>Please ensure that you have a specified a version using the version URL parameter: <code>?version=x.x.x</code></span>
        </>
      )}
    </div>
  );
 };

export default ReleaseNotesRender;
