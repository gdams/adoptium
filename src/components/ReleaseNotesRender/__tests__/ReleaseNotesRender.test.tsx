import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import userEvent from '@testing-library/user-event';
import ReleaseNotesRender from '../index';
import { fetchReleaseNotesForVersion } from '../../../hooks/fetchReleaseNotes';
import { createMockReleaseNotesAPI  } from '../../../__fixtures__/hooks';

vi.mock('../../../hooks/fetchReleaseNotes');

afterEach(() => {
    vi.clearAllMocks();
});

describe('ReleaseNotesRender component', () => {
    it('should render correctly - version not defined', () => {
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(container).toMatchSnapshot();
    });
    
    it('should render correctly', () => {
        // mock query string version
        vi.mock('query-string', () => ({
            default: {
              parse: () => ({
                version: 'version',
              }),
            }
        }));
        fetchReleaseNotesForVersion.mockReturnValue(createMockReleaseNotesAPI(1));
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(fetchReleaseNotesForVersion).toHaveBeenCalledTimes(1);
        expect(container).toMatchSnapshot();
    });

    it('pagination should work correctly', async () => {
        // mock query string version
        vi.mock('query-string', () => ({
            default: {
              parse: () => ({
                version: 'version',
              }),
            }
        }));
        fetchReleaseNotesForVersion.mockReturnValue(createMockReleaseNotesAPI(121));
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–20 of 121');
        const nextButton = screen.getByLabelText('next page');
        await userEvent.click(nextButton).then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('21–40 of 121');
        });
        const previousButton = screen.getByLabelText('previous page');
        await userEvent.click(previousButton).then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–20 of 121');
        });
        const LastButton = screen.getByLabelText('last page');
        await userEvent.click(LastButton).then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('121–121 of 121');
        });
        const FirstButton = screen.getByLabelText('first page');
        await userEvent.click(FirstButton).then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–20 of 121');
        });
        const ChangeRowsPerPage = screen.getByLabelText('rows per page');
        await userEvent.selectOptions(ChangeRowsPerPage, '50').then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–50 of 121');
        });
        // Simulate returning all rows
        await userEvent.selectOptions(ChangeRowsPerPage, '-1').then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–121 of 121');
        });
        await userEvent.selectOptions(ChangeRowsPerPage, '20').then(async() => {
            expect(container.getElementsByClassName('MuiTablePagination-displayedRows')[0].textContent).toBe('1–20 of 121');
        });
    });
});