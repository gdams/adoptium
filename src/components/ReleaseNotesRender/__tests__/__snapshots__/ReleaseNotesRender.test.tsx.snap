// Vitest Snapshot v1

exports[`ReleaseNotesRender component > should render correctly - version not defined 1`] = `
<div>
  <div>
    <h2
      class="text-center"
    >
      version
    </h2>
    <h2>
      Oops... We couldn't find those release notes
    </h2>
    <span>
      Please ensure that you have a specified a version using the version URL parameter: 
      <code>
        ?version=x.x.x
      </code>
    </span>
  </div>
</div>
`;

exports[`ReleaseNotesRender component > should render correctly 1`] = `
<div>
  <div>
    <h2
      class="text-center"
    >
      version
    </h2>
    <div
      class="p-5"
    >
      <label
        class="search-label"
        for="search-bar-0"
      >
        <span
          class="sr-only"
          id="search-bar-0-label"
        >
          Search this table
        </span>
        <input
          aria-labelledby="search-bar-0-label"
          class="form-control "
          id="search-bar-0"
          placeholder="Search"
          type="text"
          value=""
        />
      </label>
      <div
        class="react-bootstrap-table"
      >
        <table
          class="table table-striped table-hover table-bordered"
        >
          <thead>
            <tr>
              <th
                aria-label="Issue sortable"
                class="sortable"
                tabindex="0"
              >
                Issue
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Component sortable"
                class="sortable"
                tabindex="0"
              >
                Component
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Priority sortable"
                class="sortable"
                tabindex="0"
              >
                Priority
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                tabindex="0"
              >
                Title
              </th>
            </tr>
          </thead>
          <tbody
            class="text-nowrap"
          >
            <tr>
              <td>
                <a
                  href="https://link_mock/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  id_mock
                </a>
              </td>
              <td />
              <td />
              <td>
                title_mock
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="row react-bootstrap-table-pagination"
      >
        <div
          class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <span
            class="react-bs-table-sizePerPage-dropdown dropdown"
            style="visibility: visible;"
          >
            <button
              aria-expanded="false"
              class="btn btn-default btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              id="pageDropDown"
              type="button"
            >
              20
               
              <span>
                <span
                  class="caret"
                />
              </span>
            </button>
            <ul
              aria-labelledby="pageDropDown"
              class="dropdown-menu "
              role="menu"
            >
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="10"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  10
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="25"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  25
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="30"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  30
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="50"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  50
                </a>
              </li>
            </ul>
          </span>
          <span
            class="react-bootstrap-table-pagination-total"
          >
             Showing rows 
            1
             to 
            1
             of 
            1
          </span>
        </div>
        <div
          class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <ul
            class="pagination react-bootstrap-table-page-btns-ul"
          >
            <li
              class="active page-item"
              title="1"
            >
              <a
                class="page-link"
                href="#"
              >
                1
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
